pkg/api/validation/cluster_validation_test.go

```diff
@@ -40,129 +43,120 @@ func TestValidateScyllaCluster(t *testing.T) {
 	})
 
 	tests := []struct {
-		name    string
-		obj     *v1.ScyllaCluster
-		allowed bool
+		name              string
+		obj               *v1.ScyllaCluster
+		expectedErrorList field.ErrorList
 	}{
 		{
-			name:    "valid",
-			obj:     validCluster,
-			allowed: true,
+			name:              "valid",
+			obj:               validCluster,
+			expectedErrorList: field.ErrorList{},
 		},
 		{
-			name:    "two racks with same name",
-			obj:     sameName,
-			allowed: false,
+			name:              "two racks with same name",
+			obj:               sameName,
+			expectedErrorList: field.ErrorList{field.Duplicate(field.NewPath("spec", "datacenter", "racks").Key("test-rack").Child("name"), "test-rack")},
```

Comment:
i think, in this case, it would be more beneficial to compare the errors string to see if the make sense

you could compare Aggregates

Reply:
I've changed comparing ErrorLists to comparing Aggregates but using DeepEqual still requires creating actual field.Errors, i.e. errorString is not deep equal with field.Error. I don't really see any benefit in comparing aggregates instead of comparing ErrorLists, but creating field.Errors literals is probably better than using the functions. Let me know what you think

Reply:
yeah, you're right, the real type is still hidden in the error interface. My motivation is to see the error strings that will be shown to the users. It looks like we won't get it at once so how about we do `expectedErrorList` as you had before and add `expectedErrorString` field which you'd compare to `.ToAggregate.String()`?

Reply:
Changed to checking `expectedErrorList` and `expectedErrorString`